runtime/cudaq/distributed/builtin/activate_custom_mpi.sh

```diff
@@ -48,7 +48,9 @@ fi
 
 echo "Using $CXX to build the MPI plugin for MPI installation in $MPI_PATH."
 lib_mpi_plugin="$this_file_dir/libcudaq_distributed_interface_mpi.so"
+# --disable-mlir-links for https://github.com/NVIDIA/cuda-quantum/issues/2892
 $CXX -shared -std=c++17 -fPIC \
+    --disable-mlir-links \
```

Comment:
I'm curious. What is this option?
```console
% clang++ -c --disable-mlir-links marwa.cpp
clang-16: error: unsupported option '--disable-mlir-links'
%
```

Reply:
That is a good point - This is a `nvq++` only option. So if `$CXX` is, for example `g++` this will fail as is.

Reply:
Thanks for the explanation.